examples/word_language_model/learner.py

```diff
@@ -0,0 +1,92 @@
+import inferno
+import torch
+from torch.autograd import Variable
+from sklearn.metrics import f1_score
+
+class Learner(inferno.NeuralNet):
+
+    def __init__(self,
+                 criterion=torch.nn.CrossEntropyLoss,
+                 clip=0.25,
+                 lr=20,
+                 ntokens=10000,
+                 *args, **kwargs):
+        self.clip = clip
+        self.ntokens = ntokens
+        super(Learner, self).__init__(criterion=criterion, lr=lr, *args, **kwargs)
+
+    def repackage_hidden(self, h):
+        """Wraps hidden states in new Variables, to detach them from their history."""
+        if isinstance(h, Variable):
+            v = Variable(h.data)
+            return v.cuda() if self.use_cuda else v
+        else:
+            return tuple(self.repackage_hidden(v) for v in h)
+
+    def on_train_begin(self, net, *args, **kwargs):
+        self.hidden = self.module_.init_hidden(self.batch_size)
+        self.module_.cuda()
```

Comment:
Shouldn't this depend on `self.use_cuda`? And do we not need this in the base class?